apps/web/src/app/(signing)/sign/[token]/text-field.tsx

```diff
@@ -163,27 +238,69 @@ export const TextField = ({ field, recipient, onSignField, onUnsignField }: Text
       )}
 
       {!field.inserted && (
-        <p className="group-hover:text-primary text-muted-foreground text-lg duration-200">Text</p>
+        <p
+          className={cn(
+            'group-hover:text-primary text-muted-foreground flex flex-col items-center justify-center duration-200',
+            {
+              'group-hover:text-yellow-300': !field.inserted && !parsedFieldMeta?.required,
+              'group-hover:text-red-300': !field.inserted && parsedFieldMeta?.required,
+            },
+          )}
+        >
+          <span className="flex items-center justify-center gap-x-1">
+            <Type />
+            {fieldDisplayName}
+          </span>
+        </p>
       )}
 
-      {field.inserted && <p className="text-muted-foreground duration-200">{field.customText}</p>}
+      {field.inserted && (
+        <p className="text-muted-foreground dark:text-background/80 flex items-center justify-center gap-x-1 duration-200">
+          {field.customText.length < 20
+            ? field.customText
+            : field.customText.substring(0, 15) + '...'}
+        </p>
+      )}
 
       <Dialog open={showCustomTextModal} onOpenChange={setShowCustomTextModal}>
         <DialogContent>
-          <DialogTitle>
-            Enter your Text <span className="text-muted-foreground">({recipient.email})</span>
-          </DialogTitle>
-
-          <div className="">
-            <Label htmlFor="custom-text">Custom Text</Label>
+          <DialogTitle>{parsedFieldMeta?.label ? parsedFieldMeta?.label : 'Add Text'}</DialogTitle>
 
-            <Input
+          <div>
+            <Textarea
               id="custom-text"
-              className="border-border mt-2 w-full rounded-md border"
-              onChange={(e) => setLocalCustomText(e.target.value)}
+              placeholder={parsedFieldMeta?.placeholder ?? 'Enter your text here'}
+              className={cn('mt-2 w-full rounded-md', {
+                'border-2 border-red-300 ring-2 ring-red-200 ring-offset-2 ring-offset-red-200 focus-visible:border-red-400 focus-visible:ring-4 focus-visible:ring-red-200 focus-visible:ring-offset-2 focus-visible:ring-offset-red-200':
+                  userInputHasErrors,
+              })}
+              value={localText}
+              onChange={handleTextChange}
             />
           </div>
 
+          {parsedFieldMeta?.characterLimit && !userInputHasErrors && (
```

Comment:
if the characterLimit is set to 0, it will always render `0` even though `charactersRemaining` is negative like in below screenshot its -5:
![image](https://github.com/user-attachments/assets/0cdcac8b-67bf-4ddc-a4db-b7d6f7992900)

Field settings: 
![image](https://github.com/user-attachments/assets/95bc6499-4a6f-4b44-8647-ed60e3d86a80)

Reply:
Thanks for the comment & suggestions!

```
if the characterLimit is set to 0, it will always render 0 even though charactersRemaining is negative like in below screenshot its -5:
```

You're right. Observed that as well. Will update that.

```
Also it would be better to understand what the number actually is about. 
```

Regarding the 2nd point, I'm not sure what you mean. That's already on the signing page.

![third-text-field-input](https://github.com/user-attachments/assets/c11d8635-8e55-4f6b-bc5e-499d60de5a2f)

Reply:
ignore the second Point. I am stupid and during testing removed the suffix... my bad.